src/polynomials/standard-basis/immutable-polynomial.jl

```diff
@@ -106,8 +106,18 @@ function polynomial_composition(p::ImmutableDensePolynomial{B,T,X,N}, q::Immutab
     cs = evalpoly(q, p.coeffs)
     convert(P, cs)
 end
+function polynomial_composition(p::AbstractUnivariatePolynomial{B,T,X}, q::ImmutableDensePolynomial{B,S,Y,0}) where {B<:StandardBasis,T,S,X,Y}
+    P = ImmutableDensePolynomial{B,promote_type(T,S), Y,0}
+    zero(P)
```

Comment:
A polynomial composed with zero is the polynomial's constant term, not always zero. For example, if we have `f(x) = 1` and `g(x) = 0`, `f∘g` is `1`, not `0`.

Reply:
Apart from that, I don't think the promotion is necessary, that is, I think you could just ignore `S`, because the value of `q` is always zero, independently of `S`.